lib/sshkit/ssh/connection.ex

```diff
@@ -42,27 +53,53 @@ defmodule SSHKit.SSH.Connection do
     open(to_charlist(host), options)
   end
   def open(host, options) do
-    port = Keyword.get(options, :port, 22)
-    timeout = Keyword.get(options, :timeout, :infinity)
-    ssh_modules = Keyword.get(options, :ssh_modules, @ssh_modules)
-    ssh = erlang_module(%{ssh_modules: ssh_modules}, :ssh)
+    {ssh_options, sshkit_options} = fetch_options(options)
+
+    ssh = ssh_module(sshkit_options)
+    case ssh.connect(host, sshkit_options.port, ssh_options, sshkit_options.timeout) do
+      {:ok, ref} -> {
+        :ok,
+        %Connection{
+          host: host,
+          port: sshkit_options.port,
+          options: ssh_options,
+          ref: ref,
+          ssh_modules: sshkit_options.ssh_modules
+        }
+      }
+      err -> err
+    end
+  end
 
-    defaults = [user_interaction: false]
+  defp fetch_options(options) do
+    valid_sshkit_option_keys = Keyword.keys(@default_sshkit_connection_options)
 
-    options =
-      defaults
+    sshkit_options =
+      @default_sshkit_connection_options
       |> Keyword.merge(options)
-      |> Keyword.drop([:port, :timeout, :ssh_modules])
+      |> Keyword.take(valid_sshkit_option_keys)
+      |> install_dry_run_modules
+      |> Enum.into(%{})
+
+    erlang_ssh_options =
+      @default_erlang_ssh_connection_options
+      |> Keyword.merge(options)
+      |> Keyword.drop(valid_sshkit_option_keys)
       |> Utils.charlistify()
 
-    case ssh.connect(host, port, options, timeout) do
-      {:ok, ref} -> {:ok, %Connection{host: host, port: port, options: options, ref: ref, ssh_modules: ssh_modules}}
-      err -> err
+    {erlang_ssh_options, sshkit_options}
+  end
+
+  defp install_dry_run_modules(options) do
```

Comment:
I'm not sure about the naming here. It fits, but it kinda does not fit 100% 😉 

Looking at it from the outside, what this function does is a bit like *flipping the switch for dry-run if applicable*...

But let's just stick with this for now, I cannot come up with something better.